src/Version.h

```diff
@@ -0,0 +1,133 @@
+/*
+* Copyright (C) 2021 German Aerospace Center (DLR/SC)
+*
+* Created: 2021-11-02 Martin Siggel <Martin.Siggel@dlr.de>
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#ifndef VERSION_H
+#define VERSION_H
+
+#include <string>
+#include <regex>
+#include <exception>
+
+/**
+ * @brief A class to parse and compare version strings
+ */
+class Version
+{
+public:
+    explicit Version(const std::string& str)
+        : m_major(0),
+          m_minor(0),
+          m_patch(0)
+    {
+        parse(str);
+    }
+
+    int vMajor() const
+    {
+        return m_major;
+    }
+
+    int vMinor() const
+    {
+        return m_minor;
+    }
+
+    int vPatch() const
+    {
+        return m_patch;
+    }
+
+    std::string vLabel() const
+    {
+        return m_label;
+    }
+
+    std::string vBuild() const
+    {
+        return m_build;
+    }
+
+    bool operator< (const Version& other) const
```

Comment:
This is just a comment, your code looks fine. C++17 has [std::lexicographical_compare](https://en.cppreference.com/w/cpp/algorithm/lexicographical_compare) for these kinds of operators. The link contains example implementations that take only a few lines of code.

Reply:
Yes, I was saying we could copy-paste the implementation from cppreference, which is c++11 compliant. But like I said, we can also keep it as is.

Reply:
You are right! But I don't think that a rewrite has any true benefits in any terms (readabilty, simplicity, performance) except from beeing somewhat more awesome.

Reply:
😄 true